augly/image/transforms.py

```diff
@@ -188,6 +188,98 @@ def apply_transform(
         return F.apply_pil_filter(image, filter_type=self.filter_type, metadata=metadata)
 
 
+class DistortBarrel(BaseTransform):
+    def __init__(
+        self, a: float = 0.0, b: float = 0.0, c: float = 0.0, d: float = 1.0, p: float = 1
+    ):
+        """
+        @param a: Coefficient A in the equation Rsrc(r). Larger values results in more
+            barrel effect, has higher effect than b and c.
+
+        @param b: Coefficient B in the equation Rsrc(r). Larger values results in more
+            barrel effect, has lower effect than a and higher effect than c.
+
+        @param c: Coefficient C in the equation Rsrc(r). Larger values results in more
+            barrel effect, has lower effect than a and b.
+
+        @param d: Coefficient D in the equation Rsrc(r). Controls the overall scaling of
+            the image. In a positive domain, values larger than 1 will shrink the image.
+            Negative values would result in both vertically and horizontally flipped
+            image scaled in a mirrored way of positive domain.
+
+        @param p: the probability of the transform being applied; default value is 1.0
+        """
+        super().__init__(p)
+        self.a = a
+        self.b = b
+        self.c = c
+        self.d = d
+
+    def apply_transform(
+        self, image: Image.Image, metadata: Optional[List[Dict[str, Any]]] = None
+    ) -> Image.Image:
+        """
+        Applies barrel distortion to the image
+
+        @param image: PIL Image to be augmented
+
+        @param metadata: if set to be a list, metadata about the function execution
+            including its name, the source & dest width, height, etc. will be appended to
+            the inputted list. If set to None, no metadata will be appended or returned
+
+        @returns: Augmented PIL Image
+        """
+        return F.distort_barrel(
+            image, a=self.a, b=self.b, c=self.c, d=self.d, metadata=metadata
+        )
+
+
+class DistortPincushion(BaseTransform):
+    def __init__(
+        self, a: float = 0.0, b: float = 0.0, c: float = 0.0, d: float = 1.0, p: float = 1
+    ):
+        """
+        @param a: Coefficient A in the equation Rsrc(r). Larger values results in more
+        pincushion effect, has higher effect than b and c.
+
+        @param b: Coefficient B in the equation Rsrc(r). Larger values results in more
+        pincushion effect, has higher effect than b and c.
+
+        @param c: Coefficient C in the equation Rsrc(r). Larger values results in more
+        pincushion effect, has higher effect than b and c.
```

Comment:
address changes/fixes to `b` & `c` param descriptions as was potentially done in `functional.py` please!!